File: fragment/fragment/src/main/java/com/laioffer/demo2/Demo2Activity.java
package com.laioffer.demo2;

import android.os.Bundle;
import android.widget.Button;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import com.laioffer.demo1.Fragment1;
import com.laioffer.demo1.Fragment2;
import com.laioffer.fragment.fragment.R;

/**
 * 类的描述: demo2:动态添加fragment
 * Created by 春夏秋冬在中南 on 2023/6/10 02:25
 */
public class Demo2Activity extends AppCompatActivity {
  @Override
  protected void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_demo2);

    Button btn1 = findViewById(R.id.btn_show_fragment1);
    btn1.setOnClickListener(v -> {
      FragmentManager fm = getSupportFragmentManager();
      FragmentTransaction ft = fm.beginTransaction();
      Fragment1 fragment1 = new Fragment1();
      ft.add(R.id.demo2_fragment_container, fragment1);
      ft.commit();
    });

    Button btn2 = findViewById(R.id.btn_show_fragment2);
    btn2.setOnClickListener(v -> {
      FragmentManager fm = getSupportFragmentManager();
      FragmentTransaction ft = fm.beginTransaction();
      Fragment2 fragment2 = new Fragment2();
      ft.add(R.id.demo2_fragment_container, fragment2);
      ft.commit();
    });
  }
}

File: mvx/src/main/java/com/laioffer/lesson2_livedata/demo2/Demo2Activity.kt
package com.laioffer.lesson2_livedata.demo2

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.laioffer.mvx.R

class Demo2Activity : AppCompatActivity() {
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.single_button_layout)

    // 启动服务
    val button = findViewById<Button>(R.id.button)
    button.setOnClickListener {
      startService(Intent(this, MyService::class.java))
      Toast.makeText(this, "推送服务器启动成功", Toast.LENGTH_SHORT).show()
    }

    // 一股脑执行，它不会去检测生命周期
    // sendMessage(handler)


    // 眼睛 观察者  微信列表可见的情况下，才能做事情
    MyLiveData.data1.observe(this) {
      Log.e("server", "界面可见，说明用户在查看微信列表界面啦，更新消息列表UI界面:${it}")
      Toast.makeText(this, "更新消息列表UI界面成功:${it}", Toast.LENGTH_SHORT).show()
    }
  }
}